source/server/http/admin.cc

```diff
@@ -643,20 +657,26 @@ Http::Code AdminImpl::handlerAdminHome(const std::string&, Http::HeaderMap& resp
   response_headers.insertContentType().value().setReference(
       Http::Headers::get().ContentTypeValues.Html);
 
+  response.add(absl::StrReplaceAll(AdminHtmlStart, {{"@FAVICON@", EnvoyFavicon}}));
+
   // Prefix order is used during searching, but for printing do them in alpha order.
-  std::map<std::string, const UrlHandler*> sorted_handlers;
-  for (const UrlHandler& handler : handlers_) {
-    sorted_handlers[handler.prefix_] = &handler;
-  }
+  for (const UrlHandler* handler : sortedHandlers()) {
+    const std::string& url = handler->prefix_;
+
+    // For handlers that mutate state, render the link as a button in a POST form,
+    // rather than an anchor tag.  This should discourage crawlers that find the /
+    // page from accidentally mutating all the server state by GETting all the hrefs.
```

Comment:
Is there benefit in having crawlers ever traverse the links? Could we use POST for all handlers?

I actually don't know much about the zoo of crawlers out there, I would guess that for search crawlers there is less value, but maybe there are special purpose ones that might be crawling around for Envoy instances and cataloging them.

Reply:
I'm not really that concerned about what crawlers do, I just don't want them to mutate server state while crawling, should any be running on the network.  It's not just crawlers too; if a human is poking around a status page they might think twice before hitting a button, but should feel free to visit underlined links.